sdk/digitaltwins/digitaltwins/samples/javascript/dt_digitaltwins_lifecycle.js

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+const { DefaultAzureCredential } = require("@azure/identity");
+const { DigitalTwinsClient } = require("@azure/digitaltwins");
+const { v4 } = require("uuid");
+const { buildingTwin } = require("../dtdl/digitalTwins/buildingTwin.ts");
+const { building } = require("../dtdl/models/building.ts");
+
+// Scenario example of how to:
+// - create a DigitalTwins Service Client using the DigitalTwinsClient constructor
+// - create digital twin based on a model
+// - get digital twin
+// - update digital twin using JSON patch
+// - delete digital twin
+//
+// Preconditions:
+// - Environment variables have to be set
+// - DigitalTwins enabled device must exist on the ADT hub
+//
+// For the purpose of this example we will create temporary digital twin using random Ids.
+// We have to make sure these Ids are unique within the DT instance so we use generated UUIDs.
+async function main() {
+  // - AZURE_URL: The tenant ID in Azure Active Directory
+  const url = process.env.AZURE_URL;
+
+  // DefaultAzureCredential expects the following three environment variables:
+  // - AZURE_TENANT_ID: The tenant ID in Azure Active Directory
+  // - AZURE_CLIENT_ID: The application (client) ID registered in the AAD tenant
+  // - AZURE_CLIENT_SECRET: The client secret for the registered application
+  const credential = new DefaultAzureCredential();
+  const serviceClient = new DigitalTwinsClient(url, credential);
+
+  // Create model first
+  const newModels = [building];
+  const model = await serviceClient.createModels(newModels);
+  console.log(`Model: ${model.body}`);
+
+  // Create digital twin based on the created model
+  const digitalTwinId = `digitalTwin-${v4()}`;
+  const newTwin = buildingTwin;
+  const createdTwin = await serviceClient.upsertDigitalTwin(
+    digitalTwinId,
+    newTwin,
+    (enableUpdate = true)
```

Comment:
Per review: we should avoid this pattern in our code. The usual recommendation is to use option bags so you can write `{enableUpdate: true}`.

When that can't be used, it's accepted to use comments to name boolean params e.g. `doFoo(/*forceUpdate*/ true)`

Reply:
Fixed

---
In reply to: [455407628](https://github.com/Azure/azure-sdk-for-js/pull/9787#discussion_r455407628) [](ancestors = 455407628)